cumulus/client/cirrus-executor/src/bundle_processor.rs

```diff
@@ -285,6 +285,57 @@ where
 			return Ok(None)
 		}
 
+		let best_execution_chain_number = self
+			.primary_chain_client
+			.runtime_api()
+			.best_execution_chain_number(&BlockId::Hash(primary_hash))?;
+
+		let best_execution_chain_number =
+			<NumberFor<Block>>::decode(&mut best_execution_chain_number.encode().as_slice())
+				.expect("Primary number and secondary number must use the same type; qed");
+
+		assert!(
+			header_number > best_execution_chain_number,
+			"Consensus chain number must larger than execution chain number by at least 1"
+		);
+
+		// Ideally, the receipt of current block will be included in the next block, i.e., no
+		// missing receipts.
+		if header_number == best_execution_chain_number + One::one() {
+			return self.try_sign_and_send_receipt(primary_hash, execution_receipt)
+		} else {
+			// Receipts for some previous blocks are missing.
+			let max_drift = self
```

Comment:
I'm wondering what `max_drift` has to do with this though?

It is used in `pallet-executor` for a very different purpose and I don't understand why is it necessary here.

I think this `if/else + while` could have been replaced with just
```rust
for to_send in best_execution_chain_number + One::one()..=header_number
```

Reply:
If you submit a receipt that is larger than the best_execution_chain_number + max_drift, it will be immediately rejected as being too far in validate_unsigned anyway.

`T::BlockNumber` does not have the Step trait bound, so the while loop can be replaced with `..`.

Reply:
Hm, okay, makes sense.